### ISSUE TITLE
"FocusLock: could not obtain ref to internal node"

### ISSUE THREAD

**Issue by AylinCelik** [theKashey/react-focus-lock#123]
Hi, currently I'm using v2.1.0.

When I upgrade to v2.1.1 and upwards, the FocusLock works still fine.
But when I update my jest-snapshots I get the following output:
>     console.error ../base/node_modules/react-focus-lock/dist/cjs/Lock.js:133
>       FocusLock: could not obtain ref to internal node

Do I need to make adjustments while upgrading? Please help

**Comment by theKashey** [comment]
That message was [added in 2.1.1](https://github.com/theKashey/react-focus-lock/commit/af0e560966ce3dd936d6fb7f3665ecab972bdd20#diff-757809db6137e7d28cf62c382a00aeadR103-R106). You had a problem before, it was just not reported.

The problem is quite fatal actually - FocusLock needs a DOM node to work with.
If it's actually works - then it's not getting in on FocusLock mount, but later. Could it be a case?

**Comment by itutto** [comment]
I'm experiencing the same issue. The issue only happens in the test spec files. (using `react-test-renderer`)
In my test I see no reason why FocusLock would miss the DOM node.

**Comment by theKashey** [comment]
🤷‍♂️. To be solved among with #125

**Comment by AlekseyManetov** [comment]
I guess that is not solved, this error is actual with latest 2.9.1 version

**Comment by theKashey** [comment]
Sounds like I should add at least one RTL-backed test

**Comment by ortsevlised** [comment]
any updates on this issue?

**Comment by theKashey** [comment]
Unfortunately, I am still not able to reproduce this particular problem.

If you are experiencing this problem - please suggest a broken test.

**Comment by hectorohernz** [comment]
I recently ran into this error message while creating a snapshot with the "react-test-renderer" renderer. The renderer will simply convert any provided components to pure javascript objects, and without depending on the **dom**.

Since there is no dom this means our DOM node never gets attached to the **.current**, which leads us to our error statement. 

[The solution is to mock the refs. 
](https://stackoverflow.com/a/40854433/16403012)

```JAVASCRIPT 
function createNodeMock(element) {
    if (element.type === "div") {
        return document.createElement("div");
    }
}

const tree = renderer.create( <Modal / > , {
    createNodeMock
});
```

**Comment by theKashey** [comment]
So that is more synthetic use case rather than a real "bug".

I will correct this message to trigger only if a custom `container` is used - that would suppress message for all "standard" usecases, but leave it in place for moments when it can be actually useful.

**Comment by hectorohernz** [comment]
Appreciate it!

**Comment by theKashey** [comment]
Resolved in v`2.9.2`
